docs/configuration.asciidoc

```diff
@@ -0,0 +1,623 @@
+[[configuration]]
+
+ifdef::env-github[]
+NOTE: For the best reading experience,
+please view this documentation at https://www.elastic.co/guide/en/apm/agent/nodejs/current/configuration.html[elastic.co]
+endif::[]
+
+=== Configuration
+
+The available configuration options are listed below.
+Most configuration options can be set either in the optional `options` object,
+by using environment variables,
+or via the <<agent-configuration-file,agent configuration file>>.
+Their equivalent environment variable name is listed together with each option.
+
+Environment variables will always take precedence over properties of the `options` object.
+
+The only required parameter is <<service-name,`serviceName`>>.
```

Comment:
While not strictly related to this PR, I just stumbled upon this sentence that we probably overlooked when we implemented to fallback to `package.json`. So while we remember, we might as well update this text to say:

```suggestion
The only required parameter is <<service-name,`serviceName`>>.
However,
the agent will use the `name` from `package.json` by default if available.
```

Reply:
fixed.

Reply:
🤔 I don't see it anywhere. Maybe something went wrong during the rebase? 😬